### PR TITLE
[slack] Choose the correct value for the `oldest` param

### LINKED ISSUE #7987: slack: cursor.pagination_finished template logic is unclear or incorrect
While investigating an issue with the Slack integration, it was noticed that the logic for determining the state of `cursor.pagination_finished` may be incorrect, and does need clarification or correction.

> ... , it seems like the `.cursor.next_oldest_date` value should only be used if `.cursor.pagination_finished` is true, and I'm not sure how that's ever true. The `pagination_finished` value template looks like it requires *both* no `.last_response.body.response_metadata` and `.last_response.body.response_metadata.next_cursor` to be `""`:
>   ```
>   cursor:
>     pagination_finished:
>       # Use this flag to identify if an execution was interrupted in the middle
>       # of a pagination cycle.
>       # For completed pagination next_cursor is "" and if the response is not paginated
>       # then the blob response_metadata does not exist
>       value: '[[and (not (index .last_response.body "response_metadata")) (eq (.last_response.body.response_metadata.Get "next_cursor") "")]]'
>   ```

### PR DESCRIPTION
**Pull request by chrisberkhout**
## Proposed commit message

```
[slack] Choose the correct value for the `oldest` param

Choosing the correct value means:

- Use the chosen `oldest` value for all requests in a pagination
  sequence. 
- Choose the `oldest` value for a new pagination sequence based on
  results from the first page of the previous sequence.
- Save the `oldest` value in cursor data in order to facilitate
  resumption of an interrupted pagination sequence.

The `max_date_seen_so_far` request parameter is added as a workaround
for the HTTP JSON input's `cursor:` section not being able to read
existing cursor values when generating new values. It will be ignored
by the Slack API.

Another change was to use the saved `next_cursor` value to indicate
whether a pagination sequence is in progress, rather than spreading
this information across multiple values.
```

## API behavior and test scenario explanation

The system test scenario is updated here to match the recently observed behavior of the live API. It attempts to cover all the pagination logic required for non-interrupted operation.

Relevant behavior of the live API includes:
- The API needs the `oldest` parameter in order to know where to stop, even if a `cursor` parameter is given. In https://github.com/elastic/integrations/pull/8169 the `oldest` parameter was included in all requests, but further changes are required to ensure that its value is correct and does not change for a given sequence of pages.
- The `.response_metadata.next_cursor` field is always returned, as an empty string if there is no further page.
- It is common for multiple entries to have the same `date_create` timestamp. Often this is a few entries, sometimes it is many more.
- The `oldest` parameter is inclusive, so when a new sequence sets the `oldest` value to the maximum `date_create` of the previous sequence, there will be repeated results.
- Internally, a cursor identifies a particular `date_create` and `id`. The entries returned are ordered by `date_create` descending, but for a given `date_create`, the `id` values are not in order.

The new scenario is summarized in the following table:

Screenshot from 2023-10-17 09-54-13

The table's time N and identifier NNN match the system test data's timestamp suffix and ID prefix. The table shows which entry a cursor points to, but cursor values in the test data are random. Green entries are new records for ingestion, red entries are repeated records that we should avoid duplicating.

Without the new fix that scenario fails, having missed entries `3-813` and `3-212`.

Things needed for this test to pass:
- [x] Correct cursor handling (seems to work already).
- [x] The correct `oldest` value should be used for each sequence of pages.
- [x] Duplicates should be avoided (currently done by setting `_id` to the fingerprinted `id`).

<details>
<summary>Example request logs from successful system test execution</summary>

```json
{
  "@timestamp": "2023-11-02T16:59:14.095Z",
  "message": "HTTP request",
  "url.query": "limit=2&max_date_seen_so_far=0&oldest=1681664354",
  "http.response.body.content": null
}
{
  "@timestamp": "2023-11-02T16:59:14.096Z",
  "message": "HTTP response",
  "url.query": null,
  "http.response.body.content": [
    {
      "id": "4433a45b-6c7d-8900-e12f-3456789gh0i1",
      "date_create": 1683836273
    },
    {
      "id": "313b13e3-28a3-41f0-9ace-a20952def3a0",
      "date_create": 1683836273
    }
  ]
}
{
  "@timestamp": "2023-11-02T16:59:16.070Z",
  "message": "HTTP request",
  "url.query": "cursor=YXNkZmFzZGZhc2Rm&limit=2&max_date_seen_so_far=1683836273&oldest=1681664354",
  "http.response.body.content": null
}
{
  "@timestamp": "2023-11-02T16:59:16.072Z",
  "message": "HTTP response",
  "url.query": null,
  "http.response.body.content": [
    {
      "id": "80928060-1659-4b27-ad55-fdba12e3a7b1",
      "date_create": 1683836272
    },
    {
      "id": "1885fb41-c67c-4cf5-a5c4-d90cb58dd5f9",
      "date_create": 1683836271
    }
  ]
}
{
  "@timestamp": "2023-11-02T16:59:26.077Z",
  "message": "HTTP request",
  "url.query": "limit=2&max_date_seen_so_far=1683836273&oldest=1683836273",
  "http.response.body.content": null
}
{
  "@timestamp": "2023-11-02T16:59:26.080Z",
  "message": "HTTP response",
  "url.query": null,
  "http.response.body.content": [
    {
      "id": "4216a45b-6c7d-8900-e12f-3456789gh0i1",
      "date_create": 1683836274
    },
    {
      "id": "412d13e3-28a3-41f0-9ace-a20952def3a0",
      "date_create": 1683836273
    }
  ]
}
{
  "@timestamp": "2023-11-02T16:59:26.081Z",
  "message": "HTTP request",
  "url.query": "cursor=GytjmKHF5hFmty&limit=2&max_date_seen_so_far=1683836274&oldest=1683836273",
  "http.response.body.content": null
}
{
  "@timestamp": "2023-11-02T16:59:26.083Z",
  "message": "HTTP response",
  "url.query": null,
  "http.response.body.content": [
    {
      "id": "81328070-1659-4b27-ad55-fdba12e3a7b1",
      "date_create": 1683836273
    },
    {
      "id": "2125fb41-c67c-4cf5-a5c4-d90cb58dd5f9",
      "date_create": 1683836273
    }
  ]
}
{
  "@timestamp": "2023-11-02T16:59:26.084Z",
  "message": "HTTP request",
  "url.query": "cursor=Hi4JrvZZnX3IGHE1&limit=2&max_date_seen_so_far=1683836274&oldest=1683836273",
  "http.response.body.content": null
}
{
  "@timestamp": "2023-11-02T16:59:26.085Z",
  "message": "HTTP response",
  "url.query": null,
  "http.response.body.content": [
    {
      "id": "4433a45b-6c7d-8900-e12f-3456789gh0i1",
      "date_create": 1683836273
    },
    {
      "id": "313b13e3-28a3-41f0-9ace-a20952def3a0",
      "date_create": 1683836273
    }
  ]
}
{
  "@timestamp": "2023-11-02T16:59:36.075Z",
  "message": "HTTP request",
  "url.query": "limit=2&max_date_seen_so_far=1683836274&oldest=1683836274",
  "http.response.body.content": null
}
{
  "@timestamp": "2023-11-02T16:59:36.077Z",
  "message": "HTTP response",
  "url.query": null,
  "http.response.body.content": [
    {
      "id": "22328080-1659-4b27-ad55-fdba12e3a7b1",
      "date_create": 1683836275
    },
    {
      "id": "4216a45b-6c7d-8900-e12f-3456789gh0i1",
      "date_create": 1683836274
    }
  ]
}
{
  "@timestamp": "2023-11-02T16:59:46.075Z",
  "message": "HTTP request",
  "url.query": "limit=2&max_date_seen_so_far=1683836275&oldest=1683836275",
  "http.response.body.content": null
}
{
  "@timestamp": "2023-11-02T16:59:46.077Z",
  "message": "HTTP response",
  "url.query": null,
  "http.response.body.content": [
    {
      "id": "6465fc41-c67c-4cf5-a5c4-d90cb58dd5f9",
      "date_create": 1683836276
    },
    {
      "id": "22328080-1659-4b27-ad55-fdba12e3a7b1",
      "date_create": 1683836275
    }
  ]
}
{
  "@timestamp": "2023-11-02T16:59:56.076Z",
  "message": "HTTP request",
  "url.query": "limit=2&max_date_seen_so_far=1683836276&oldest=1683836276",
  "http.response.body.content": null
}
{
  "@timestamp": "2023-11-02T16:59:56.077Z",
  "message": "HTTP response",
  "url.query": null,
  "http.response.body.content": [
    {
      "id": "6465fc41-c67c-4cf5-a5c4-d90cb58dd5f9",
      "date_create": 1683836276
    }
  ]
}
```
</details>

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

```
# build and test
elastic-package build -v
elastic-package test system -v

# find agent logs and copy to current directory
docker exec elastic-package-stack-elastic-agent-1 find /usr/share/elastic-agent/state/data/logs -type f |\
  xargs -I{} docker cp elastic-package-stack-elastic-agent-1:{} .

# print request log fields relevant to pagination
cat http-*.ndjson | jq '{"@timestamp",message,"url.query","http.response.body.content": (try (."http.response.body.content" | fromjson | .entries|map({id,date_create})) catch null)}'
```

## Related issues

- Closes #7987

### CODE REVIEW

**Comment by efd6** [comment]
> The request parameter `max_date_seen_so_far` is intended to be ignored by the API, and is added because the in the HTTP JSON input's `cursor:` section it is not possible to read existing cursor values.

Do we know that Slack's API will tolerate this param being present?

**Review by bhapas — COMMENTED** [review]
I think we should test this against a slack instance to see if the API tolerates the new param

**Comment by efd6** [comment]
@chrisberkhout Thanks for checking that. Would you mind including a note in the commit message that this technique is used to stash the value?

**Review by efd6 — APPROVED** [review]

File: packages/slack/manifest.yml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+version: "1.15.1"
 description: "Slack Logs Integration"
 type: integration
 categories:
   - productivity
   - security
 conditions:
   kibana:
-    version: "^8.7.1"
+    version: "^8.10.1"
```
[the remark this answers is not in the document]
Comment (chrisberkhout) [reply packages/slack/manifest.yml:12]: Support for the `max` function in value templates was added in 8.10.0 https://github.com/elastic/beats/pull/36036

**Pull request merged**